src/webview_provider/newDocsGenPanel.ts

```diff
@@ -130,147 +126,118 @@ export class NewDocsGenPanel
-          this.sendResponseToWebview({
-            command: "response",
-            data: {
+        this.handleSyncRequestFromWebview(
+          syncRequestId,
+          async () => {
+            const columnsFromSources = await this.queryManifestService
+              .getProject()
+              ?.getColumnsOfSource(args.source as string, args.table as string);
+            return {
               columns: columnsFromSources
                 ? columnsFromSources.map((c) => c.column)
                 : [],
-            },
-            syncRequestId,
-          });
-        } catch (error) {
-          this.dbtTerminal.error(
-            "newDocsGenPanel:getColumnsOfSources",
-            "unable to get columns of sources",
-            error,
-          );
-          const message =
-            error instanceof PythonException
-              ? error.exception.message
-              : (error as Error).message;
-          window.showErrorMessage(extendErrorWithSupportLinks(message));
-          this.sendResponseToWebview({
-            command: "response",
-            data: { columns: [] },
-            syncRequestId,
-          });
-        }
+            };
+          },
+          command,
+          true,
+        );
         break;
+
       case "getColumnsOfModel":
-        try {
-          const columns = await this.queryManifestService
-            .getProject()
-            ?.getColumnsOfModel(args.model as string);
-          this.sendResponseToWebview({
-            command: "response",
-            data: {
+        this.handleSyncRequestFromWebview(
+          syncRequestId,
+          async () => {
+            const columns = await this.queryManifestService
+              .getProject()
+              ?.getColumnsOfModel(args.model as string);
+            return {
               columns: columns ? columns.map((c) => c.column) : [],
-            },
-            syncRequestId,
-          });
-        } catch (error) {
-          this.dbtTerminal.error(
-            "newDocsGenPanel:getColumnsOfModel",
-            "unable to get columns of models",
-            error,
-          );
-          const message =
-            error instanceof PythonException
-              ? error.exception.message
-              : (error as Error).message;
-          window.showErrorMessage(extendErrorWithSupportLinks(message));
-          this.sendResponseToWebview({
-            command: "response",
-            data: { columns: [] },
-            syncRequestId,
-          });
-        }
+            };
+          },
+          command,
+          true,
+        );
         break;
+
       case "getSourcesInProject":
-        const sources = this.queryManifestService.getSourcesInProject(
-          window.activeTextEditor?.document.uri,
-        );
+        this.handleSyncRequestFromWebview(
+          syncRequestId,
+          async () => {
+            const sources = this.queryManifestService.getSourcesInProject(
```

Comment:
get project from message?

Reply:
all these commands are triggered based on the documentation of current model that is open in editor. do we still need to handle that from message?